soc/arm/nordic_nrf/nrf52/power.c

```diff
@@ -82,38 +32,63 @@ void sys_set_power_state(enum power_states state)
 void sys_power_state_post_ops(enum power_states state)
 {
 	switch (state) {
-	case SYS_POWER_STATE_CPU_LPS:
-	case SYS_POWER_STATE_CPU_LPS_1:
-		/* Enable interrupts */
-		__set_BASEPRI(0);
-		break;
-#if defined(CONFIG_SYS_POWER_DEEP_SLEEP)
+#ifdef CONFIG_SYS_POWER_DEEP_SLEEP
+# ifdef CONFIG_SYS_POWER_STATE_DEEP_SLEEP_SUPPORTED
 	case SYS_POWER_STATE_DEEP_SLEEP:
+		/* Nothing to do. */
 		break;
+# endif
 #endif
 	default:
-		/* Unsupported State */
-		LOG_ERR("Unsupported State");
+		LOG_ERR("Unsupported power state %u", state);
 		break;
 	}
+
+	/*
+	 * System is now in active mode. Reenable interrupts which were disabled
+	 * when OS started idling code.
+	 */
+	__set_BASEPRI(0);
 }
 
 bool sys_is_valid_power_state(enum power_states state)
 {
 	switch (state) {
+#ifdef CONFIG_SYS_POWER_LOW_POWER_STATE
+# ifdef CONFIG_SYS_POWER_STATE_CPU_LPS_SUPPORTED
```

Comment:
why keep lps support here when it was removed. Commit message states: `- Removed LPS modes`

Reply:
It's probably better not to use a SoC that does not support LPS mode to showcase LPS mode. If anything the example should show that it works also with a SoC that does not support all the power modes.

Reply:
@nordic-krch: No. I think that it is the effect of the development history. I have suggested @ramakrishnapallala, who develops the power management subsystem, to make these definitions generic.

Reply:
> SoC may not support given LPS mode but application may, e.g. dimming screen or LEDs.

I don't quite understand the logic of the sentence. If an application supports SPI transfer but SoC doesn't should the SoC report that it supports SPI transfer?

The main problem with `sys_is_valid_power_state` function is that power subsystem is using it without providing any description (and neither a declaration). Looking at other comments in this PR e.g.
> it looks like this function is generic. It's output depends only on generic kconfig options. Why not move it to the subsystem?

 it looks like we don't really know what this function is supposed to be doing.

The description in zephyr/soc/arm/nordic_nrf/nrf52/soc_power.h says however "Check a low power state is supported by SoC". And in this case the implementation provided in this PR is wrong. We really should clean it up.

Reply:
@mnkp: This is way I asked @ramakrishnapallala, who designed power management, to join this discussion.